dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/patch/factories/impl/PatchOperation.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository.patch.factories.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.dspace.app.rest.exception.PatchBadRequestException;
+import org.dspace.app.rest.model.RestModel;
+import org.dspace.app.rest.model.patch.Operation;
+import org.springframework.data.rest.webmvc.json.patch.LateObjectEvaluator;
+/**
+ * This patch class includes abstract and implemented methods that
+ * can be used to type check objects derived from operation values.
+ *
+ * TODO: Determine whether this type checking is needed for resource patch operations.
+ *
+ * @author Michael Spalti
+ */
+public abstract class PatchOperation<R extends RestModel, T extends Object>
+        implements ResourcePatchOperation<R> {
+
+    public abstract RestModel perform(R resource, Operation operation)
+            throws PatchBadRequestException;
+
+    /**
+     * Throws PatchBadRequestException for missing operation value.
+     * @param value the value to test
+     */
+    public void checkOperationValue(T value) {
+        if (value == null) {
+            throw new PatchBadRequestException("No value provided for the operation.");
+        }
+    }
+
+    // This is duplicated code (see org.dspace.app.rest.submit.factory.impl.PatchOperation)
+    // If it stays here, it should be DRY. Current patch resource patch operations do not
+    // use these methods.
+    public List<T> evaluateArrayObject(LateObjectEvaluator value) {
+        List<T> results = new ArrayList<T>();
+        T[] list = null;
+        if (value != null) {
+            LateObjectEvaluator object = (LateObjectEvaluator) value;
+            list = (T[]) object.evaluate(getArrayClassForEvaluation());
+        }
+
+        for (T t : list) {
+            results.add(t);
+        }
+        return results;
+    }
+
+    public T evaluateSingleObject(LateObjectEvaluator value) {
```

Comment:
Could you add Javadocs here please, if this method is used/need.

Reply:
Will add javadoc if method is needed!

Reply:
Added javadoc. It's not showing up here in the diff but view file and it's there. ...?